golang/tracing.go

```diff
@@ -92,12 +92,17 @@ func (s Span) TracingEnabled() bool { return (s.flags & tracingFlagEnabled) == t
 
 // NewChildSpan begins a new child span in the provided Context
 func (s Span) NewChildSpan() *Span {
-	return &Span{
+	childSpan := &Span{
 		traceID:  s.traceID,
 		parentID: s.spanID,
-		spanID:   uint64(rng.Int63()),
 		flags:    s.flags,
 	}
+	if s.spanID == 0 {
```

Comment:
This is fine, but isn't technically a requirement according to the <a href="https://twitter.github.io/zipkin/Instrumenting.html#generating-identifiers">zipkin docs</a>.  The only requirement for the initial child span is that it have the same trace id as the parent, and that its parent id be the span id of the parent.  Of course if we want to impose this additional correlation requirement, I guess that's fine.